segmentstore/server/src/main/java/io/pravega/segmentstore/server/logs/Throttler.java

```diff
@@ -138,13 +140,12 @@ boolean isThrottlingRequired() {
     }
 
     private CompletableFuture<Void> throttleOnce(ThrottlerCalculator.DelayResult delay) {
-        this.metrics.processingDelay(delay.getDurationMillis());
         if (delay.isMaximum()
                 || delay.getThrottlerName() == ThrottlerCalculator.ThrottlerName.DurableDataLog) {
             // Increase logging visibility if we throttle at the maximum limit (which means we're likely to fully block
             // processing of operations) or if this is due to us not being able to ingest items quickly enough.
             log.warn("{}: Processing delay = {}.", this.traceObjectId, delay);
-        } else {
+        } else if (delay.getThrottlerName() != null) {
```

Comment:
Why would this be null? 

And even if it is, I **still** want to see this logged somewhere - this is important for certain debugging activities.

Reply:
I did not dig too deeply, but It appears that `throttle` is being called somewhere in the testing code, without having been assigned a `Throttler` and returns a `DelayResult` with a default `null` name. When I called `delay.getThrottlerName()` on this null value, it halts execution. This was the reason for my TravisCI builds failing. Having said that I think this condition is redundant anyways. Will remove.

Reply:
Well, by looking at the code in this `else` block I don't see how that could happen. `log.debug` is passed `delay`, and it will only invoke `toString` on it if debug-level logging is enabled. I believe it is not (but I am not sure). However if there is a bug like you said, then it must be in the `toString` implementation of `delay`, so if anything it should be fixed there.